test/jest_integration/basic_auth.test.ts

```diff
@@ -86,4 +89,51 @@ describe('start kibana server', () => {
       .unset('Authorization');
     expect(response.status).toEqual(401);
   });
+
+  it('call authinfo API with cookie', async () => {
+    const authCookie = await getAuthCookie(root, ADMIN_USER, ADMIN_PASSWORD);
+
+    const response = await kbnTestServer.request
+      .get(root, '/api/v1/auth/authinfo')
+      .unset(AUTHORIZATION_HEADER_NAME)
+      .set('Cookie', authCookie);
+    expect(response.status).toEqual(200);
+  });
+
+  it('login with user name and password', async () => {
+    const response = await kbnTestServer.request
+      .post(root, '/auth/login')
+      .unset(AUTHORIZATION_HEADER_NAME)
+      .send({
+        username: ADMIN_USER,
+        password: ADMIN_PASSWORD,
+      });
+    expect(response.status).toEqual(200);
+    const authCookie = extractAuthCookie(response);
+    // cookie is like security_authentication=<cookie>, verify there are actual cookie value
+    expect(authCookie.split('=')[1]).toBeTruthy();
+  });
+
+  it('login fails with invalid credentials', async () => {
+    const response = await kbnTestServer.request
+      .post(root, '/auth/login')
+      .unset(AUTHORIZATION_HEADER_NAME)
+      .send({
+        username: ADMIN_USER,
+        password: 'invalid',
+      });
+    expect(response.status).toEqual(401);
+  });
+
```

Comment:
Is the anonymous login going to be tested?

Reply:
sure, will write test case for anonymous auth